### PR TITLE
Adding support for async tests

### PR DESCRIPTION
**Pull request by chrisjshull**
Issue: https://github.com/elwayman02/ember-sinon-qunit/issues/55

Covers both returning a Promise and using `assert.async()`.

Drive-by:
Updated file location to `addon-test-support/` so that addons and apps can import the same way (did this as a non-breaking change - there is just a console warning if you import in the older style).

### CODE REVIEW

**Comment by elwayman02** [comment]
@chrisjshull this looks awesome! At first glance it looks good, will give it another pass later. Would also love thoughts from @MiguelMadero!

File: README.md [hunk 1]
```diff
@@ -6,16 +6,16 @@
 [![Code Climate](https://codeclimate.com/github/elwayman02/ember-sinon-qunit/badges/gpa.svg)](https://codeclimate.com/github/elwayman02/ember-sinon-qunit)
 [![Codacy Badge](https://api.codacy.com/project/badge/8c6fbb028801423fbd4b1bfe17c9b1a0)](https://www.codacy.com/app/hawker-jordan/ember-sinon-qunit)
 
-This addon integrates [Sinon](http://jhawk.co/sinonjs) & [Ember-QUnit](http://jhawk.co/ember-qunit) 
+This addon integrates [Sinon](http://jhawk.co/sinonjs) & [Ember-QUnit](http://jhawk.co/ember-qunit)
 via [Ember-Sinon](http://jhawk.co/ember-sinon), as inspired by [Sinon-QUnit](http://jhawk.co/sinon-qunit).
 
 Why not simply use Ember-Sinon? Two reasons:
 
-1. Ember-Sinon does not handle cleanup of Ember-QUnit tests. Ember-Sinon-QUnit provides a test method
```
Comment (MiguelMadero) [inline README.md]: minor: it's usually desired to preserve whitespace on .md files. That's the default in `.editorconfig`, I suggest configuring your editor to follow this, it makes it easier to contribute to other people's projects, which respecting their conventions.

**Comment by MiguelMadero** [comment]
Oh wow, this ended up being more involved than what I was signing up for. Thanks for addressing this!

File: README.md (hunk 1 above)
Comment (chrisjshull) [inline README.md]: odd, my editor is supposed to respect `.editorconfig` out of the box. Thanks for pointing that out, I think I have it fixed now.

As it doesn't have a material impact here, I'm tempted to leave the change - is that ok?

Comment (elwayman02) [inline README.md]: I would prefer to revert, as the whitespace at the end of the lines is in fact intentional on my part.

Comment (chrisjshull) [inline README.md]: whitespace now put back

**Pull request merged**